### PR TITLE
chore: `aft version-bump` test suite

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Description of changes:*
- Add new version bump test suite which runs against a snapshot of the Amplify Flutter mono repo

This PR re-writes the existing test suite for `aft version-bump`. A description of this test suite can be found [here](https://github.com/aws-amplify/amplify-flutter/blob/chore/version-bump-tests/packages/aft/test/version_bump/README.md).

All but ~300 LOC added in this PR are generated. The generated changes are separated into a single commit. To view changes without that commit see https://github.com/aws-amplify/amplify-flutter/pull/5424/files/c481d8f2d7c0555873ee816dcd8b058d99134f4f

Note: This PR does not attempt to fix any of the existing issues in `aft version-bump`. The diffs in this PR represent the output of `aft version-bump` as of this PR, which means there are known defects in the diffs. For example, the diff [here](https://github.com/aws-amplify/amplify-flutter/pull/5424/commits/9b87b5979f71c9285a3d72cf588a983729175437#diff-4cb71c49ccfc42dbaed4864e275e4dc120b63430b1cdfec0cf7183b69268379fR65-R69) shows that `amplify_analytics_pinpoint_dart` had dependency version changes, but the package version was not updated. This is a defect that needs to be addressed in a follow up PR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/aft/test/version_bump/version_bump_test.dart [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+Future<void> copyDirectory(
+  Directory source,
+  Directory destination,
+) async {
+  for (final e in source.listSync(recursive: true)) {
+    if (!File(e.path).existsSync()) {
+      continue;
+    }
+    final path = e.path.replaceFirst(source.path, '');
+    logger.info('copying to ${destination.path}/$path');
```
Comment (Equartey) [inline packages/aft/test/version_bump/version_bump_test.dart:212]:
```suggestion
    logger.verbose('copying to ${destination.path}/$path');
```
nit: this is cluttering the test output and I don't see the value to always include.

File: packages/aft/test/version_bump/version_bump_test.dart [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+      '$title\n',
+    ]);
+    return runGit(repo, ['rev-parse', 'HEAD']);
+  }
+}
+
+Future<String> runGit(Repo repo, List<String> args) async {
+  final result = await repo.git.runCommand(args);
+  final stdout = result.stdout as String;
+  logger.info('git ${args.join(' ')}:\n$stdout');
```
Comment (Equartey) [inline packages/aft/test/version_bump/version_bump_test.dart:199]:
```suggestion
  logger.verbose('git ${args.join(' ')}:\n$stdout');
```
Same here

File: packages/aft/test/version_bump/version_bump_test.dart [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+          if (generateSnapshots) {
+            final process = await repo.git.runCommand(['diff', '--unified=0']);
+            final diff = process.stdout as String;
+            await diffFile.writeAsString(processDiff(diff));
+          }
+          await tempDirectory.delete(recursive: true);
+        });
+
+        group('chore', () {
+          test('single amplify package', () async {
```
Comment (Equartey) [inline packages/aft/test/version_bump/version_bump_test.dart:140]: Q: Is `single amplify package` an accurate description for all the tests?

Comment (Jordan-Nelson) [reply packages/aft/test/version_bump/version_bump_test.dart]: No, I should have updated that

**Review by Equartey — APPROVED** [review]

**Review by tyllark — APPROVED** [review]

**Pull request merged**